src/app/modules/rup/components/core/rup.component.ts

```diff
@@ -284,4 +284,19 @@ export class RUPComponent implements OnInit, AfterViewInit {
         const hasValue = !!this.registro.valor;
         return !hasValue;
     }
+
+    /**
+     * Costrasta los requeridos contra los registros para determinar exactamente sobre que iterar.
+     */
+    get requeridos() {
+        const requeridos = [];
+        for (let i = 0; i < this.registro.registros.length; i++) {
+            const concepto = this.registro.registros[i].concepto;
+            const requerido = this.elementoRUP.requeridos[i];
+            if (requerido && requerido.concepto.conceptId === concepto.conceptId) {
```

Comment:
Aquí habría que comparar cada uno de los registros con cada uno de los requeridos, porque una vez que uno no coincide, ya el índice no coincide entonces no se muestran los demás items que si se deberían mostrar. Es decir si tenía 3 registros, pero el del medio no es requerido, entonces se va a mostrar el primero por el tercer registro se guarda como segundo requerido. Entonces el tercer registro en este for no tiene con que compararse y no ingresa.

Reply:
Quedo aplicado el cambio.